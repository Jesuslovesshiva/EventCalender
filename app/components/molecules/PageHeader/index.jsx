import classNames from 'classnames'
import PropTypes from 'prop-types'

import useTranslation from '@/hooks/use-translation'
import customPropTypes from '@/utils/prop-types'

const PageHeader = props => {
  const title = useTranslation(props.title)

  return (
    <div
      className={classNames(
        'grid grid-cols-3 items-center px-6 py-4 bg-gradient-to-b bg-gradient-silver shadow',
        props.className
      )}
    >
      <div className="flex items-center">
        <h1 className="shrink-0 grow-0 mr-3 font-page-heading text-primary">
          {title}
        </h1>
        {props.leftSlot}
      </div>
      <div className="justify-self-center">{props.centerSlot}</div>
      <div className="justify-self-end">{props.rightSlot}</div>
    </div>
  )
}

PageHeader.propTypes = {
  centerSlot: PropTypes.node,
  className: PropTypes.string,
  leftSlot: PropTypes.node,
  rightSlot: PropTypes.node,
  title: customPropTypes.Translatable.isRequired
}

PageHeader.defaultProps = {
  centerSlot: null,
  className: null,
  leftSlot: null,
  rightSlot: null
}

export default PageHeader
