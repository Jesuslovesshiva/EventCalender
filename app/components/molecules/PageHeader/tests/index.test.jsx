import render from '@/utils/test-utils/custom-render'

import PageHeader from '..'

describe('PageHeader', () => {
  it('should match the snapshot', () => {
    const {container} = render(
      <PageHeader
        title="Title"
        leftSlot="left"
        centerSlot="center"
        rightSlot="right"
      />
    )
    expect(container.firstChild).toMatchSnapshot()
  })
})
