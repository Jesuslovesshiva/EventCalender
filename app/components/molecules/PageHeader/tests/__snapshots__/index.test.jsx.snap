// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageHeader should match the snapshot 1`] = `
<div
  class="grid grid-cols-3 items-center px-6 py-4 bg-gradient-to-b bg-gradient-silver shadow"
>
  <div
    class="flex items-center"
  >
    <h1
      class="shrink-0 grow-0 mr-3 font-page-heading text-primary"
    >
      Title
    </h1>
    left
  </div>
  <div
    class="justify-self-center"
  >
    center
  </div>
  <div
    class="justify-self-end"
  >
    right
  </div>
</div>
`;
